rust/diesel/src/order.rs

```diff
@@ -50,10 +50,9 @@ pub struct ProductOrder {
 }
 
 impl Order {
-    pub fn create(order: NewUserOrder, connection: &PgConnection) -> Option<Order> {
+    pub fn create(order: NewUserOrder, connection: &PgConnection) -> Result<Order, String> {
```

Comment:
Nit: you can use `Box<dyn Error>` as your error type if you want to keep the type generic without converting all errors into strings, then just use `.map_err(|err| err.into())` instead of `.map_err(|err| err.to_string())`. It doesn't make much of a difference here, but it is good practice.

Same thing with users and products.

Reply:
I agree, it is a good practice , but I would argue to leave it as is due to simplicity.
If you think it is a must, I'll change this part

Reply:
That makes sense, I think it is fine to leave it as-is then.